src/window.c

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#include "window.h"
+#include "memory.h"
+
+// This function prepares to draw in a window by setting up its boundary
+// restrictions.
+// THIS FUNCTION SHOULD ALWAYS BE CALLED BEFORE MAKING A GRAPHICS DRAW
+// CALL IN THE WINDOW SYSTEM
+static inline void window_begin_draw(struct window *w) {
+    graphics_set_bounds(w->x, w->y, w->width, w->height);
+}
+
+// This function removes the restrictions used to enforce window clipping
+// on the graphics system
+// THIS FUNCTION SHOULD ALWAYS BE CALLED IF window_begin_draw WAS CALLED
+// TO FINISH THE DRAW
+static inline void window_end_draw() {
+    graphics_clear_bounds();
+}
+
+struct window *window_create(int x, int y, int width, int height) {
+    struct window *w = kmalloc(sizeof(*w));
+
```

Comment:
I know that this is harmless because graphics will never draw to it, given that graphics will never set its bounds outside of video_res, but the way the code is implies that we can create windows outside of the bounds of the screen. Just something we need to be aware of.